### PR TITLE
Reduce mem usage for CI with valgrind

### PR DESCRIPTION
**Pull request by rustyrussell**
This should reduce the flakes we're getting due to kill -9, which I suspect is OOM killer.

I printed out the total %mem, mem and vsz every 5 seconds and ran the full testsuite with VALGRIND=1 SLOW_MACHINE=1 (non-parallel).  The largest mem users were those creating many nodes (of course), and this should help with that.

Changelog-None

### CODE REVIEW

File: contrib/pyln-testing/pyln/testing/utils.py [hunk 1]
```diff
@@ -690,11 +691,14 @@ def __init__(self, node_id, lightning_dir, bitcoind, executor, valgrind, may_fai
         if dsn is not None:
             self.daemon.opts['wallet'] = dsn
         if valgrind:
+            trace_pattern = '*python*,*bitcoin-cli*,*elements-cli*'
+            if valgrind_plugins is False:
```
Comment (cdecker) [inline contrib/pyln-testing/pyln/testing/utils.py:695]: `if not valgrind_plugins`

File: contrib/pyln-testing/pyln/testing/utils.py [hunk 2]
```diff
@@ -690,11 +691,14 @@ def __init__(self, node_id, lightning_dir, bitcoind, executor, valgrind, may_fai
...
+                trace_pattern += ',*plugins*'
```
Comment (cdecker) [inline contrib/pyln-testing/pyln/testing/utils.py:696]: Maybe this should be called `trace_skip_pattern`

**Review by cdecker — APPROVED** [review]
Just some naming nits and CI is sad due to a lint, but looking good otherwise

File: contrib/pyln-testing/pyln/testing/utils.py [hunk 3]
```diff
@@ -1280,12 +1284,20 @@ def get_nodes(self, num_nodes, opts=None):
 
         assert len(opts) == num_nodes
 
+        # Only trace one random node's plugins, to avoid OOM.
+        if SLOW_MACHINE:
+            valgrind_plugins = [False] * num_nodes
+            valgrind_plugins[random.randint(0, num_nodes-1)] = True
```
Comment (vincenzopalazzo) [inline contrib/pyln-testing/pyln/testing/utils.py:1290]:
```suggestion
            valgrind_plugins[random.randint(0, num_nodes - 1)] = True
```

I suspect that this is the cause of the CI failure (sadly)

**Review by vincenzopalazzo — APPROVED** [review]
LGTM

Today I was looking in some tools for Github action that prints some env information at the end of the test. This could be useful I think, but there are nothings around.

**Comment by rustyrussell** [comment]
Thanks, pushed those fixes!

Hope this will unstick a heap of other PRs

**Review by ZmnSCPxj — APPROVED** [review]
ACK 7f6276461115dcc02679ce3f00c5c31b9e6634cd

**Review by vincenzopalazzo — APPROVED** [review]
ack https://github.com/ElementsProject/lightning/pull/4935/commits/7f6276461115dcc02679ce3f00c5c31b9e6634cd

**Pull request merged**